engine/src/main/java/org/hibernate/search/bridge/impl/ExtendedBridgeProvider.java

```diff
@@ -58,5 +58,13 @@ public FieldBridge provideFieldBridge(BridgeProviderContext bridgeProviderContex
 		 * not.
 		 */
 		boolean isExplicitlyMarkedAsNumeric();
+
+		/**
+		 * Returns the type of the indexed member/property; it works for arrays and collections too.
+		 *
+		 * @see #getReturnType()
+		 * @return the type of the indexed member
+		 */
+		Class<?> getExactReturnType();
```

Comment:
I know BridgeProviderContext is SPI and can't be changed, but this naming scheme scares me. It feels like `getReturnType` is somewhat random (not "exact")...

I see two solutions:

 * we could deprecate the old `getReturnType` and use more appropriate naming, like for instance `getPropertyType` and `getPropertyElementType`
 * or we could change the behavior of this method slightly, rename it `getContainerType` and make it return null when the property is not a container.

What do you think?